src/GtkObservables.jl

```diff
@@ -5,28 +5,33 @@ if isdefined(Base, :Experimental) && isdefined(Base.Experimental, Symbol("@compi
 end
 
 using LinearAlgebra   # for `inv`
-using Gtk, Colors, FixedPointNumbers, Reexport
+using Gtk4, Colors, FixedPointNumbers, Reexport
 @reexport using Observables
 using Graphics
 using Graphics: set_coordinates, BoundingBox
 using IntervalSets, RoundingIntegers
 # There's a conflict for width, so we have to scope those calls
 import Cairo
 
-using Gtk: GtkWidget
+using Gtk4: GtkScaleLeaf, GtkCheckButtonLeaf, GtkToggleButtonLeaf,
+    GtkButtonLeaf, GtkSpinButtonLeaf, GtkColorButtonLeaf,
+    GtkEntryLeaf, GtkTextViewLeaf, GtkComboBoxTextLeaf,
+    GtkLabelLeaf, GtkProgressBarLeaf
 # Constants for event analysis
-using Gtk.GConstants.GdkModifierType: SHIFT, CONTROL, MOD1
-using Gtk.GConstants.GdkScrollDirection: UP, DOWN, LEFT, RIGHT
-using Gtk.GdkEventType: BUTTON_PRESS, DOUBLE_BUTTON_PRESS, BUTTON_RELEASE
-
-if !isdefined(Gtk, :signal_handler_is_connected)
-    signal_handler_is_connected(w::GObject, handler_id::Culong) =
-        ccall((:g_signal_handler_is_connected, Gtk.libgobject), Cint, (Ptr{GObject}, Culong), w, handler_id) == 1
-end
+const SHIFT = Gtk4.ModifierType_SHIFT_MASK
+const CONTROL = Gtk4.ModifierType_CONTROL_MASK
+const MOD1 = Gtk4.ModifierType_ALT_MASK
+const BUTTON_PRESS = Gtk4.EventType_BUTTON_PRESS
+const BUTTON_RELEASE = Gtk4.EventType_BUTTON_RELEASE
+const MOTION_NOTIFY = Gtk4.EventType_MOTION_NOTIFY
+const UP = Gtk4.ScrollDirection_UP
+const DOWN = Gtk4.ScrollDirection_DOWN
+const LEFT = Gtk4.ScrollDirection_LEFT
+const RIGHT = Gtk4.ScrollDirection_RIGHT
 
 # Re-exports
 export set_coordinates, BoundingBox, SHIFT, CONTROL, MOD1, UP, DOWN, LEFT, RIGHT,
-       BUTTON_PRESS, DOUBLE_BUTTON_PRESS, destroy
```

Comment:
What's the replacement for double-click?

Reply:
In GTK4 it's one of the arguments in the signal handler.